src/forms/inputs/date-input.js

```diff
@@ -55,26 +54,44 @@ const defaultProps = {
 /* Ignore test coverage: */
 /* istanbul ignore next */
 
+// React datepicker expects a date value
+function parseDate(value) {
+  if (!value || value instanceof Date) return value
+
+  return parseISO(value)
+}
+
 function DateInput (props) {
   const {
     input: { name, value, onBlur, onChange },
     meta, // eslint-disable-line no-unused-vars,
     className, // eslint-disable-line no-unused-vars
     ...rest
   } = omitLabelProps(props)
-  const momentValue = value ? moment(value) : null
+  const dateValue = parseDate(value)
+  const calendarRef = useRef()
+
   return (
     <LabeledField { ...props }>
-      <DatePicker 
-        {...{ 
-          id: name,
-          name,
-          selected: momentValue,
-          onBlur: () => onBlur(value),
-          onChange: (val) => onChange(val ? val.toDate() : ''),
-          ...rest
-        }}
-      />
+      <div className="date-input-wrapper">
+        <DatePicker 
+          {...{ 
+            id: name,
+            name,
+            ref: calendarRef,
+            selected: dateValue || null, // passing in an empty string will not default the tabbable element to today
+            onBlur: () => onBlur(value),
+            onChange: (val) => onChange(val ?? ''),
+            onSelect: () => {
+              // onBlur isn't called after the user has selected an option. This isn't ideal because many form libraries validate onBlur.
+              // This is a temporary workaround which manually refocuses the element so that blur is triggered as expected.
+              // https://github.com/Hacker0x01/react-datepicker/issues/2028
+              setTimeout(() => calendarRef.current.setFocus(), 0) // deferFocusInput accomplishes this, but it's likely eliminated during tree shaking since it's never used internally
```

Comment:
Hmm. Using  Storybook, when I select a date (i.e., explicitly choose a day value from the little calendar that pops up) I see:
```
date input changed: Fri Jun 03 2022 00:00:00 GMT-0500 (CDT)
```
...in the "Actions" window. I only see:
```
blurred: Thu Jun 09 2022 00:00:00 GMT-0500 (CDT)
```
... when I click anywhere outside the input (after the calendar closes). Is this the expectation? 
See: https://share.getcloudapp.com/9Zuo8OpQ

Reply:
Good. It seemed correct to me too. But reading the comments in the code, I was under the impression that you expected the blur event to be generated when the date was selected (also given that the `onSelect()` prop was the one that was updated) and that **isn't** happening. Maybe clarify the comment to say that this workaround is required so that a blur event is generated _after the date input loses focus_.

Reply:
For sure. If it reads as confusing to you, it'll be the same for others! Thanks for the feedback